framework/test/unit/modules/interoperability/sidechain/commands/mainchain_registration.spec.ts

```diff
@@ -0,0 +1,461 @@
+		senderPublicKey: publicKey,
+		nonce: BigInt(0),
+		fee: BigInt(100000000),
+		params: encodedTransactionParams,
+		signatures: [publicKey],
+	});
+	const networkIdentifier = Buffer.from(
+		'e48feb88db5b5cf5ad71d93cdcd1d879b6d5ed187a36b0002cc34e0ef9883255',
+		'hex',
+	);
+	let mainchainRegistrationCommand: MainchainRegistrationCommand;
+	let verifyContext: CommandVerifyContext<MainchainRegistrationParams>;
+
+	beforeEach(() => {
+		mainchainRegistrationCommand = new MainchainRegistrationCommand(
+			MODULE_ID_INTEROPERABILITY,
+			new Map(),
+			new Map(),
+		);
+	});
+
+	describe('verify', () => {
+		beforeEach(() => {
+			verifyContext = testing
+				.createTransactionContext({
+					transaction,
+					networkIdentifier,
+				})
+				.createCommandVerifyContext<MainchainRegistrationParams>(mainchainRegParams);
+		});
+
+		it('should return status OK for valid params', async () => {
+			const result = await mainchainRegistrationCommand.verify(verifyContext);
+			expect(result.status).toBe(VerifyStatus.OK);
+		});
+
+		it('should return error if own chain id is greater than maximum uint32 number', async () => {
+			verifyContext.params.ownChainID = MAX_UINT32 + 1;
+			const result = await mainchainRegistrationCommand.verify(verifyContext);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error).toBeInstanceOf(LiskValidationError);
+		});
+
+		it('should return error if bls key is not 48 bytes', async () => {
+			verifyContext.params.mainchainValidators[1].blsKey = getRandomBytes(47);
+			const result = await mainchainRegistrationCommand.verify(verifyContext);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error).toBeInstanceOf(LiskValidationError);
+		});
+
+		it('should return error if name is greater than max length of name', async () => {
+			verifyContext.params.ownName = getRandomBytes(21).toString('hex');
+			const result = await mainchainRegistrationCommand.verify(verifyContext);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error).toBeInstanceOf(LiskValidationError);
+		});
+
+		it('should return error if name is invalid', async () => {
+			verifyContext.params.ownName = '*@#&$_2';
+			const result = await mainchainRegistrationCommand.verify(verifyContext);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error?.message).toInclude(
+				`Sidechain name is in an unsupported format: *@#&$_2`,
+			);
+		});
+
+		it('should return error if number of mainchain validators is not equal to number of mainchain validators', async () => {
+			verifyContext.params.mainchainValidators.pop();
+			const result = await mainchainRegistrationCommand.verify(verifyContext);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error).toBeInstanceOf(LiskValidationError);
+		});
+
+		it('should return error if bls keys are not lexicographically ordered', async () => {
+			const temp = verifyContext.params.mainchainValidators[0].blsKey;
```

Comment:
Optional: You can also swap like this so you don't have to use a temp variable: [verifyContext.params.mainchainValidators[0], verifyContext.params.mainchainValidators[1]] = [verifyContext.params.mainchainValidators[1], verifyContext.params.mainchainValidators[0]];